### ISSUE TITLE
The processing method in stage1

### ISSUE THREAD

**Issue by Chensihao-7** [mahmoodlab/TITAN#29]
Dear author, hello. Thank you very much for the outstanding work proposed by your team. However, I have a question: May I ask, in the work of stage1, the article mentioned that CLAM was used for processing first, then clustering was carried out to obtain the "tissue group", then the regions that did not meet the 16*16patches were screened out, and finally 16*16patches were randomly cropped out. 1. How is the clustering algorithm implemented in this (which method), and then how to filter out the tissues that do not meet the requirements? \ n2. For the last one, when randomly clipping 16*16 regions of interest, is it clipping only one on one slice or one on one "tissue group"? Thank you very much for your answer.

**Comment by sophiajw** [comment]
Dear @Chensihao-7, thank you for your interest in our work!
Here a more detailed overview of our preprocessing workflow:
1. We preprocess the WSI to patches including their coordinates (using CLAM) 
2. Based on the coordinates we construct a grid of features 
3. We group tissue regions that are close and consider distant tissue regions as separate groups to minimise background patches (specifically, if bounding boxes of tissue regions overlap, we consider them as one tissue group, no clustering algorithm involved).
4. We discard all groups with fewer than 16 patches in total.
5. We sample per tissue group. Typically, one WSI corresponds to one group but if it contains multiple distant regions (eg multiple slices) it can consist of multiple groups. 

Let me know if anything remains unclear!
Best
Sophia

**Comment by Chensihao-7** [comment]
@sophiajw Thank you very much for your reply.
I am very interested in learning from your work on fine-tuning TITAN with iBOT during Stage 1.
May I ask, during the Stage 1 training phase, did you replace the model within the iBOT GitHub framework to train your own model, or did you modify the TITAN framework directly?
I have tried to reproduce this part myself, but I found that some hyperparameter settings caused my loss not to decrease properly.
Therefore, I would like to ask for your advice.
If it is convenient, would you be willing to share the relevant part of the code?
